common/djangoapps/student/models.py

```diff
@@ -59,6 +59,26 @@
 AUDIT_LOG = logging.getLogger("audit")
 SessionStore = import_module(settings.SESSION_ENGINE).SessionStore  # pylint: disable=invalid-name
 
+UN_ENROLLED_TO_ALLOWED_TO_ENROLL = 'from unenrolled to allowed to enroll'
+ALLOWED_TO_ENROLL_TO_ENROLL = 'from allowed to enroll to enrolled'
+ENROLL_TO_ENROLL = 'from enrolled to enrolled'
+ENROLL_TO_UN_ENROLL = 'from enrolled to unenrolled'
+UN_ENROLL_TO_ENROLL = 'from unenrolled to enrolled'
+ALLOWED_TO_ENROLL_TO_UN_ENROLL = 'from allowed to enroll to enrolled'
+UN_ENROLL_TO_UN_ENROLL = 'from unenrolled to unenrolled'
+DEFAULT_TRANSITION_STATE = 'N/A'
+
+TRANSITION_STATES = (
+    (UN_ENROLLED_TO_ALLOWED_TO_ENROLL, ALLOWED_TO_ENROLL_TO_ENROLL),
+    (ALLOWED_TO_ENROLL_TO_ENROLL, ALLOWED_TO_ENROLL_TO_ENROLL),
+    (ENROLL_TO_ENROLL, ENROLL_TO_ENROLL),
+    (ENROLL_TO_UN_ENROLL, ENROLL_TO_UN_ENROLL),
+    (UN_ENROLL_TO_ENROLL, UN_ENROLL_TO_ENROLL),
+    (ALLOWED_TO_ENROLL_TO_UN_ENROLL, ALLOWED_TO_ENROLL_TO_UN_ENROLL),
+    (UN_ENROLL_TO_UN_ENROLL, UN_ENROLL_TO_UN_ENROLL),
+    (DEFAULT_TRANSITION_STATE, DEFAULT_TRANSITION_STATE)
+)
```

Comment:
I hate to say this because you probably have a number of places to update this, but reading something like ALLOWED_TO_ENROLL_TO_ENROLL is confusing. 

Maybe if it's something more like:

ALLOWEDTOENROLL_TO_ENROLL

As it makes the ALLOWEDTOENROLL to mean a particular state.

Likewise:

ALLOWEDTOENROLL_TO_UNENROLL

Where the 'TO' word separates two states that have no spacing in them?